### PR TITLE
PatchtTST and PatchTSMixer fixes

### PR DESCRIPTION
**Pull request by wgifford**
# What does this PR do?

Makes PatchTST and PatchTSMixer interfaces more consistent -- using similar parameter names for method arguments and returned data objects.

Fixes a few minor bugs in PatchTST implementation.

Ensures more consistent output shapes with regression when an output_distribution is chosen (in both forward and generate methods).

Fixes slow tests for PatchTST.

### CODE REVIEW

**Comment by amyeroberts** [comment]
Hi @wgifford, thanks for opening this PR! 

Let us know when it's ready for review!

**Comment by wgifford** [comment]
Hi @amyeroberts I just rebased on current main. I think it is ready for review.

**Review by amyeroberts — COMMENTED** [review]
Thanks for working on this! 

Overall looks good - I have some questions about the changes in outputs occurring. cc @kashif for a second review to make sure this is OK. 

This is effectively a breaking change. As the model was just added in last release and both PatchTST and PatchTSMixer have a relatively small number of downloads ~500 in the past month, I'd say this is OK cc @namctin @ajati  - this will break current pipelines, but is easy to fix (`future_values` -> `target_values`) and `prediction_outputs` -> `regression_outputs`) and enables some modules to be loaded in using the `Auto` classes. 

Have you run the slow tests for both the models to make sure they're passing? 

In the PR description, it says "Fixes slow tests for PatchTST" - were they broken before the changes in the PR, or is this updating to reflect the changes in this PR?

**Comment by kashif** [comment]
thanks @amyeroberts I'll fix it up and push suggestions

**Comment by wgifford** [comment]
> thanks @amyeroberts I'll fix it up and push suggestions

Hi Kashif, I think all the suggestions are good. Please reach out if any issues.

File: tests/models/patchtst/test_modeling_patchtst.py
Comment (wgifford) [reply tests/models/patchtst/test_modeling_patchtst.py:381]: This issue arrises because the randomly generated samples when using the GPU (slow tests) are different from those on CPU (underlying RNG is different). I believe the original values may have been from CPU.

File: tests/models/patchtsmixer/test_modeling_patchtsmixer.py
Comment (wgifford) [reply tests/models/patchtsmixer/test_modeling_patchtsmixer.py:690]: Yes, for prediction `future_values` is used.

**Comment by wgifford** [comment]
@kashif I see this is passing -- any other issues to address before we merge?

**Review by kashif — APPROVED** [review]

**Comment by wgifford** [comment]
@amyeroberts Can this be merged? Some of these fixes are critical.

**Review by amyeroberts — COMMENTED** [review]
Thanks for iterating on this! 

Just a few last things to address then we're good to merge

File: tests/models/patchtsmixer/test_modeling_patchtsmixer.py
Comment (wgifford) [reply tests/models/patchtsmixer/test_modeling_patchtsmixer.py:265]: Hi @amyeroberts depending on the model type, different arguments are required. My understanding is that this was done for consistency. Here is a quick summary of the important args:
```
Pretraining:
        past_values: torch.Tensor,
        observed_mask: Optional[torch.Tensor]

Prediction:
        past_values: torch.Tensor,
        observed_mask: Optional[torch.Tensor]
        future_values: Optional[torch.Tensor]

Classification:
        past_values: torch.Tensor,
        target_values: torch.Tensor

Regression:
        past_values: torch.Tensor,
        target_values: torch.Tensor
```

Comment (wgifford) [reply tests/models/patchtsmixer/test_modeling_patchtsmixer.py:264]: @amyeroberts I have updated lines 410-440 to make the `expected_arg_names` much more explicit and clear.

Comment (amyeroberts) [reply tests/models/patchtsmixer/test_modeling_patchtsmixer.py:264]: `expected_arg_names` looks a lot cleaner 🤗 

Here, in this if/else branch popping `"future_values"` for every branch is an indication there's something wrong. Instead `._prepare_for_class` should prepare the correct inputs for the input model_class, including the labels

Comment (wgifford) [reply tests/models/patchtsmixer/test_modeling_patchtsmixer.py]: Okay -- updated `._prepare_for_class` to create the `future_values` and/or `target_values` as needed. In the `ModelTester` class we only prepare what is truly common -- no more popping.

**Review by amyeroberts — COMMENTED** [review]
Thanks for iterating on this - LGTM!

**Review by amyeroberts — APPROVED** [review]

**Comment by wgifford** [comment]
> Thanks for iterating on this - LGTM!

Thank you @amyeroberts!

**Pull request merged**
